docs/entity_lock.md

```diff
@@ -3,14 +3,96 @@ title: Lock Entity
 sidebar_label: Lock
 ---
 
-> This entry is incomplete. Contribution welcome.
+A lock entity is a device which is able to lock and unlock. Locking and unlocking can optionally be secured with a user code. Some locks also allow for opening of latches, this may also be secured with a user code.
 
 ## Properties
 
 > Properties should always only return information from memory and not do I/O (like network requests). Implement `update()` or `async_update()` to fetch data.
 
 | Name | Type | Default | Description
 | ---- | ---- | ------- | -----------
+| changed_by | string | None | Describes what the last change was triggered by.
+| code_format | string | None | Regex for code format or None if no code is required.
+| is_locked | bool | None | Indication of whether the lock is currently locked. Used to determine `state`.
+
+### States
+| Constant | Description
+|----------|------------------------|
+| `STATE_LOCKED` | The lock is locked.
```

Comment:
No need to add this. The base class will use the value of `is_locked` to determine what is going on .

Reply:
Removed states section, as all can be understood via ```is_locked```.